lib/checks/aria/aria-allowed-attr-evaluate.js

```diff
@@ -27,21 +28,56 @@ import { getRole, allowedAttr, validateAttr } from '../../commons/aria';
  */
 function ariaAllowedAttrEvaluate(node, options, virtualNode) {
   const invalid = [];
-
   const role = getRole(virtualNode);
   const attrs = virtualNode.attrNames;
   let allowed = allowedAttr(role);
-
   // @deprecated: allowed attr options to pass more attrs.
   // configure the standards spec instead
   if (Array.isArray(options[role])) {
     allowed = uniqueArray(options[role].concat(allowed));
   }
 
+  // TODO: look into memoizing getRole and removing this local cache
```

Comment:
Why is there a todo in yer PR?

Reply:
I'll leave that for Steve to decide, but if we don't, we should have an issue open for it instead of leaving it as a "todo" in the code.

Reply:
I'm good with an issue instead of a TODO. I'll add this spot in the code in the issue so we know to remove the cache if we find we should memoize the function.

Reply:
https://github.com/dequelabs/axe-core/issues/3170